### ISSUE TITLE
PHP7: Array to string conversion errors in BaseDinklyDataModel

### ISSUE THREAD

**Issue by DannySchick** [lewsid/dinkly#38]
Loading any page that requires a data model will throw Array to String conversion notices and eventually crash

This is because of a new variable syntax. 

````
                        // old meaning            // new meaning
$foo->$bar['baz']       $foo->{$bar['baz']}       ($foo->$bar)['baz'] 
````

In hydrate(), update() and insert(), anything using the $foo->$bar[$property] syntax has to be updated to $foo->{$bar[$property]}

**Comment by lewsid** [comment]
What version of PHP are you getting this error?

On Fri, Jul 8, 2016 at 2:16 PM, Danny Schick <notifications@github.com>
wrote:

> Loading any page that requires a data model will throw Array to String
> conversion notices and eventually crash
>
> This is because of a new variable syntax.
>
>                         // old meaning            // new meaning
> $foo->$bar['baz']       $foo->{$bar['baz']}       ($foo->$bar)['baz']
>
> In hydrate(), update() and insert(), anything using the
> $foo->$bar[$property] syntax has to be updated to $foo->{$bar[$property]}
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/lewsid/dinkly/issues/38>, or mute the thread
> <https://github.com/notifications/unsubscribe/AAJ9dzy179XqeGYEBScnRWMa_2tlRybIks5qTpP1gaJpZM4JIRSf>
> .
>



-- 
Christopher Lewis
lewsid@lewsid.com
(802) 448-2149

**Comment by DannySchick** [comment]
PHP 7. Andrew told me to go through and check for any errors after upgrading

**Comment by lewsid** [comment]
Fixed in b311ab7
